adagrams/game.py

```diff
@@ -1,13 +1,101 @@
 from random import randint
-
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
 def draw_letters():
-    pass
+    pool = LETTER_POOL.copy()
+    list_letter = []
+    List_10 = []
+    for letter, count in pool.items():
+      list_letter += [letter] * count
+
+    print(list_letter)
+
+    for i in range(10):
+        random_index2 = randint(1, len(list_letter)) - 1
+        print(f"take index {random_index2}. current length {len(list_letter)}")
+        List_10.append(list_letter[random_index2])
+        list_letter.pop(random_index2) 
+    return List_10
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper() #to make sure all the letter that Player provide become uppercase.  
+    second_letterbank = letter_bank.copy()
+    for j in word:
+        if j in second_letterbank:
+            second_letterbank.remove(j) #to makesure delete the same value already appeared in word and second letbank
+        else:
+            return False
+    return True
+
+
+def score_word(word: str):
+    
+    score_chart = {
+        'A': 1, 'B': 3, 'C': 3, 'D': 2, 'E': 1, 'F': 4, 'G': 2, 'H': 4,
+        'I': 1, 'J': 8, 'K': 5, 'L': 1, 'M': 3, 'N': 1, 'O': 1, 'P': 3,
+        'Q': 10, 'R': 1, 'S': 1, 'T': 1, 'U': 1, 'V': 4, 'W': 4, 'X': 8,
+        'Y': 4, 'Z': 10
+    }
+    word_upper = word.upper()
+    point = 0
+    for letter in word_upper:
+        point = point + score_chart.get(letter) # get value of each letter in wordupper compare to word in dic, if same then sum up
+
+    if len(word_upper) >= 7 and len(word_upper) <= 10:
+        point = point + 8
+    return point
+            
+
+def get_highest_word_score(word_list): #word líst now is word_upper wave 2
+    max_score_tuple = ("", 0)
+    for word in word_list:
+        point = score_word(word)
+        
+        (max_word, max_score) = max_score_tuple
+        
+        if max_score < point:
+            max_score_tuple = (word,point)
+        elif max_score == point:
+            if len(word) == len(max_word):
+                continue
+            
+            if len(max_word) == 10:
+                continue
+            
+            if len(max_word) < len(word) and len(word) != 10:
+                continue
+                
+            max_score_tuple = (word, point)
```

Comment:
You're doing a great job of thinking through the scenarios where we don't want to update the `max_score_tuple`! 

If we can invert our thinking to describe only the scenarios where we want the code to change, we can reduce a fair bit of repetition and avoid needing to use `continue`. When we see repeated or very similar code within a function, it's often a sign that we should refactor and maybe approach the problem from another direction. My personal approach is that if I notice I've written the same thing 3 times within a function, I try something different like create a variable or see if I can change how I'm thinking about comparisons.

How could you rewrite the comparisons for tie breaking to describe only the situations where we want `max_score_tuple` to change?